R/readSource.R

```diff
@@ -146,6 +146,6 @@ readSource <- function(type, subtype = NULL, convert = TRUE) {
   x <- .getData(type, subtype, prefix)
   on.exit(toolendmessage(startinfo, "-"))
   x <- clean_magpie(x)
-  setwd(cwd)
+  local_dir(cwd)
```

Comment:
This is probably not needed here, right?

Reply:
I have no idea why, but the remind preprocessing broke because of this... I used git bisect and some trial and error to identify exactly this as the reason, but I still wonder why.

Reply:
Okay, I took a deeper look into this. `withr` does not work together with `on.exit`. `withr` adds the code to be executed after the function to an attribute of the current environment and sets an on.exit handler that reads that attribute and runs the code. In this case `on.exit` overwrote (is that a word?) the on.exit handler with its own code. Setting `local_dir` at the end had an effect, because it was setting the `withr` on.exit handler again, so that the cleanup code of the first `local_dir` was re-enabled.

Reply:
tldr: Always use `withr::defer` instead of `on.exit` when using any `withr` functions.

Reply:
What do you think of deprecating `on.exit` in our linter settings and suggesting `withr::defer` as an alternative?